src/Processors/Transforms/LightShufflingTransform.cpp

```diff
@@ -138,10 +138,12 @@ void LightShufflingTransform::consume(Chunk chunk)
 {
     if (chunk.hasRows())
     {
+        auto chunk_ctx = chunk.getChunkContext(); /// save chunk context before split
         auto split_chunks{chunk_splitter.split(chunk)};
         for (auto & chunk_with_shard : split_chunks)
         {
             assert(chunk_with_shard.chunk);
+            chunk_with_shard.chunk.setChunkContext(chunk_ctx ? chunk_ctx->clone() : nullptr);
```

Comment:
Do we need clone or shared chunk context works fine ?

Reply:
Yes, the current implementation  of WatermarkTransformWithSubstream/AggregatingTransformWithSubstream may further modify the ChunkContext. So we cannot share it in shuffled multiple threads